### ISSUE TITLE
Comment suppresses cop

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#15447]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Other

### Cop name

IfUnlessModifier

### Steps to reproduce the problem

```ruby
# "if" gets "IfUnlessModifier"
if ARGV
  2
end || 3

# the "-" comment erroneously suppresses the "IfUnlessModifier" cop
if ARGV # -
  2
end || 3
```

### Expected behavior

The comment should make no difference.

### Actual behavior

The comment makes a difference.

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
System Version: macOS 26.5.1 (25F80)
Model Identifier: MacBookAir10,1
ruby 4.0.5 (2026-05-20 revision 64336ffd0e) +PRISM [arm64-darwin25]
rubocop     2026-07-10 5e56132d8
```

**Comment by forthrin** [comment]
Not a bug?

**Comment by koic** [comment]
This is intended behavior, not a bug.

The suppression was deliberately introduced in #8499 (RuboCop 1.3). When the if keyword line has a comment and code follows the end keyword on the same line, converting to the modifier form used to produce broken code:

```ruby
# before autocorrection
if ARGV # -
  2
end || 3

# after autocorrection (broken: `|| 3` gets commented out)
(2 if ARGV) # - || 3
```

Since the modifier form collapses everything into a single line, there is no deterministic place to relocate the comment without changing the meaning of the code. In such cases, RuboCop's design is to not register an offense at all rather than to register one that cannot be safely autocorrected.

So the comment does make a difference here, by design. Closing as intended behavior.

**Comment by forthrin** [comment]
Thanks for the historic recap. Two clarifying questions:

1. Why can't it rewrite as follows? (What am I missing here? Is this a matter of technical difficulty?)

```ruby
(2 if ARGV) || 3 # -
ARGV ? 2 : 3 # -
```

2. As an alternative to saying nothing and making ones like the undersigned think it's a bug, how about something like the (admittedly quirky) message "(Re)move comment to see suppressed issue."?
